docs/concepts.md

```diff
@@ -1,173 +1,213 @@
-routed to any number of handlers, including zero. Timeout messages are always
-routed back to the handler that produced them.
+- An **event** message is a fact — an action your application has already taken.
 
-Dogma defines four handler types, one each for [aggregates](#aggregate),
-[processes](#process), [integrations](#integration) and
-[projections](#projection). These concepts are described in more detail below.
+  For example, _10 widgets were added to Alex's shopping cart_.
 
-## Application
+- A **timeout** message is a delayed request — an action that should be taken later.
 
-An **application** is a collection of [message handlers](#message-handler) that
-work together as a unit. Typically, each application encapsulates a specific
-business (sub-)domain or "bounded-context".
+  For example, _send Alex a reminder about their incomplete purchase after 24 hours_.
 
-Each application is represented by an implementation of the
-[`dogma.Application`] interface.
+Every distinct action — such as "add item to cart" or "complete purchase" — is
+represented by a different **message type**. Each message type is a Go type that
+implements one of the [`Command`], [`Event`], or [`Timeout`] interfaces.
 
-## Engine
+> [!IMPORTANT]
+> Messages must contain all the information necessary to act on them. Notice how
+> each example above explicitly references the subject — Alex — rather than an
+> implied "current user". This approach "captures intent" at the time the
+> message is created, leaving no room for ambiguity when it is handled.
 
-An engine is a Go module that delivers messages to an
-[application](#application) and persists the application's state.
+## Handlers
 
-A Dogma application can run on any Dogma engine. The choice of engine brings
-with it a set of guarantees about how the application behaves, for example:
+A **message handler** is a component of your application that acts upon
+[messages] it receives by producing new messages, updating state, or interacting
+with external systems.
 
-- **Consistency**: Different engines may provide different levels of
-  consistency guarantees, such as [immediate consistency] or [eventual
-  consistency].
+There are four types of handler:
 
-- **Persistence**: The engine may offer a choice of persistence mechanisms for
-  application state, such as in-memory, on-disk, or in a remote database.
+- An **aggregate message handler** manages a group of related entities, such
+  as a shopping cart and the items within it. It handles commands by recording
+  events that reflect changes to the entities' state. Each aggregate in your
+  application has many **instances** — for example, a shopping cart aggregate
+  might use an instance per customer. Aggregates are the main building block of
+  your application's logic.
 
-- **Scalability**: The engine may provide a choice of scalability models, such
-  as single-node or multi-node.
+  > [!NOTE]
```

Comment:
This `[!NOTE]` syntax doesn't seem to work inside a list.

Reply:
Yep, I see. It works in my VS Code preview, but not on GitHub itself. I'll move it to the end of that entire section.